app/src/main/java/com/github/bsideup/liiklus/Application.java

```diff
@@ -17,9 +26,102 @@ public static void main(String[] args) throws Exception {
     }
 
     public static ConfigurableApplicationContext start(String[] args) {
-        val application = new SpringApplication(Application.class);
-        application.setDefaultProperties(Collections.singletonMap("spring.profiles.active", "exporter,gateway"));
+        return createSpringApplication(args).run(args);
+    }
+
+    public static SpringApplication createSpringApplication(String[] args) {
+        val environment = new StandardEnvironment();
+        environment.setDefaultProfiles("exporter", "gateway");
+        environment.getPropertySources().addFirst(new SimpleCommandLinePropertySource(args));
+
+        val pluginsDir = environment.getProperty("plugins.dir", String.class, "./plugins");
+        val pathMatcher = environment.getProperty("plugins.pathMatcher", String.class, "*.jar");
+
+        log.info("Loading plugins from {} with matcher: {}", pluginsDir, pathMatcher);
+
+        val pluginManager = new DefaultPluginManager(Paths.get(pluginsDir).toAbsolutePath().normalize()) {
+
+            @Override
+            protected PluginRepository createPluginRepository() {
+                return new LiiklusPluginRepository(getPluginsRoot(), pathMatcher);
+            }
+
+            @Override
+            protected ExtensionFinder createExtensionFinder() {
+                return new LiiklusExtensionFinder(this);
+            }
+
+            @Override
+            protected PluginLoader createPluginLoader() {
+                return new DefaultPluginLoader(this, pluginClasspath) {
+
+                    @Override
+                    protected void loadClasses(Path pluginPath, PluginClassLoader pluginClassLoader) {
+                        // Don't
+                    }
+
+                    protected void loadJars(Path pluginPath, PluginClassLoader pluginClassLoader) {
+                        pluginClassLoader.addFile(pluginPath.toFile());
+                        try (FileSystem jarFileSystem = FileSystems.newFileSystem(pluginPath, null)) {
+                            for (String libDirectory : this.pluginClasspath.getLibDirectories()) {
+                                Path libPath = jarFileSystem.getPath(libDirectory);
+                                if (Files.exists(libPath)) {
+                                    Files.walk(libPath, 1).filter(it -> !Files.isDirectory(it)).forEach(it -> {
+                                        try {
+                                            Path tempFile = Files.createTempFile(it.getFileName().toString(), ".jar");
+                                            Files.copy(it, tempFile, StandardCopyOption.REPLACE_EXISTING);
+                                            pluginClassLoader.addURL(tempFile.toUri().toURL());
+                                        } catch (Exception e) {
+                                            e.printStackTrace();
```

Comment:
meh :|

Reply:
I remember fixing it o_O anyways, fixed now :)